src/features/hotspots/details/HotspotDetails.tsx

```diff
@@ -265,190 +295,239 @@ const HotspotDetails = ({
     )
   }, [hotspot?.address, t])
 
+  const cardHandle = useCallback(() => {
+    return (
+      <HotspotSheetHandle hotspot={hotspot} toggleSettings={toggleSettings} />
+    )
+  }, [hotspot, toggleSettings])
+
+  const handleHeaderLayout = (event: LayoutChangeEvent) => {
+    const nextSnapPoints = [event.nativeEvent.layout.height, hp(55)]
+    onLayoutSnapPoints?.({
+      collapsed: nextSnapPoints[0],
+      expanded: nextSnapPoints[1],
+    })
+    setSnapPoints(nextSnapPoints)
+  }
+
+  const handleChange = useCallback(
+    (index: number) => {
+      setListIndex(index)
+      onChangeHeight(snapPoints[index])
+    },
+    [onChangeHeight, snapPoints],
+  )
+
+  const handleToggleStatus = useCallback(async () => {
+    if (listIndex === 0) {
+      listRef.current?.snapTo(1)
+      if (showStatusBanner) {
+        return // banner is already showing, but was out of sight
+      }
+    }
+    if (listIndex === 0) {
+      await sleep(300) // Add a little delay to avoid animation jank
+    }
+    toggleShowStatusBanner()
+  }, [listIndex, showStatusBanner, toggleShowStatusBanner])
+
   if (!hotspot) return null
 
   return (
-    <BottomSheetScrollView keyboardShouldPersistTaps="always">
-      <Box paddingBottom="l">
-        <Box onLayout={onLayoutHeader}>
-          <Box marginBottom="lm" alignItems="flex-start" marginHorizontal="m">
-            <Text
-              variant="light"
-              fontSize={29}
-              lineHeight={31}
-              color="black"
-              textAlign="center"
-              numberOfLines={1}
-              adjustsFontSizeToFit
-            >
-              {formattedHotspotName[0]}
-            </Text>
-            <Box flexDirection="row" alignItems="flex-start">
+    <BottomSheet
+      snapPoints={snapPoints}
+      ref={listRef}
+      index={0}
```

Comment:
One thing that might be nice is to retain the snap index. So if for example you select a hotspot from your list and expand the details, when you hit the home icon to go back to the list and select another the card is in the last state you had it (eg it remains expanded or collapsed)

Not sure if that makes sense or if its intended to reset everytime?

Reply:
I hear what you're saying. Let's see what others think before making the change.